modules/ktx/texture_loader_ktx.cpp

```diff
@@ -0,0 +1,561 @@
+					break;
+			}
+			break;
+		}
+		default:
+			ktxTexture_Destroy(ktx_texture);
+			ERR_FAIL_V_MSG(Ref<Resource>(), "Unsupported version KTX texture file '" + p_path + "'.");
+			break;
+	}
+
+	Vector<uint8_t> src_data;
+
+	/* KTX use 4-bytes padding, don't use mipmaps if padding is effective
+	 * XXX unpad dynamically
+	 */
+	int pixel_size = Image::get_format_pixel_size(format);
+	int pixel_rshift = Image::get_format_pixel_rshift(format);
+	int block = Image::get_format_block_size(format);
+	int minw, minh;
+	Image::get_format_min_pixel_size(format, minw, minh);
+	int w = width;
+	int h = height;
+	for (uint32_t i = 0; i < mipmaps; ++i) {
+		ktx_size_t mip_size = ktxTexture_GetImageSize(ktx_texture, i);
+		size_t bw = w % block != 0 ? w + (block - w % block) : w;
+		size_t bh = h % block != 0 ? h + (block - h % block) : h;
+		size_t s = bw * bh;
+		s *= pixel_size;
+		s >>= pixel_rshift;
+		if (mip_size != static_cast<ktx_size_t>(s)) {
+			if (!i) {
+				ktxTexture_Destroy(ktx_texture);
+				ERR_FAIL_V_MSG(Ref<Resource>(), "Unsupported padded KTX texture file '" + p_path + "'.");
+			}
+			mipmaps = 1;
+			break;
+		}
+		w = MAX(minw, w >> 1);
+		h = MAX(minh, h >> 1);
+	}
+
+	for (uint32_t i = 0; i < mipmaps; ++i) {
+		ktx_size_t mip_size = ktxTexture_GetImageSize(ktx_texture, i);
+		ktx_size_t offset;
+		if (ktxTexture_GetImageOffset(ktx_texture, i, 0, 0, &offset) != KTX_SUCCESS) {
+			ktxTexture_Destroy(ktx_texture);
+			ERR_FAIL_V_MSG(Ref<Resource>(), "Invalid KTX texture file '" + p_path + "'.");
+		}
+		int prev_size = src_data.size();
+		src_data.resize(prev_size + mip_size);
+		memcpy(src_data.ptrw() + prev_size, ktxTexture_GetData(ktx_texture) + offset, mip_size);
+	}
+
+	Ref<Image> img = memnew(Image(width, height, mipmaps - 1, format, src_data));
+	if (srgb)
+		img->srgb_to_linear();
+
+	if (r_error) {
+		*r_error = OK;
+	}
+
+	ktxTexture_Destroy(ktx_texture);
+	return img;
+}
+
+Ref<Resource> ResourceFormatKTX::load(const String &p_path, const String &p_original_path, Error *r_error, bool p_use_sub_threads, float *r_progress, CacheMode p_cache_mode) {
+	if (r_error) {
+		*r_error = ERR_CANT_OPEN;
+	}
+
+	Error err;
+	Ref<FileAccess> f = FileAccess::open(p_path, FileAccess::READ, &err);
+	if (f.is_null()) {
+		return Ref<Resource>();
+	}
+
+	Ref<FileAccess> fref(f);
+	if (r_error) {
+		*r_error = ERR_FILE_CORRUPT;
+	}
+
+	ERR_FAIL_COND_V_MSG(err != OK, Ref<Resource>(), "Unable to open KTX texture file '" + p_path + "'.");
+	Ref<Image> img = load_from_file_access(f, r_error, p_path);
+	Ref<ImageTexture> texture = ImageTexture::create_from_image(img);
+	return texture;
+}
+
+static Ref<Image> _ktx_mem_loader_func(const uint8_t *p_ktx, int p_size) {
+	Ref<FileAccessMemory> f;
+	f.instantiate();
+	f->open_custom(p_ktx, p_size);
+	Error err;
+	Ref<Image> img = load_from_file_access(f, &err, "memory");
```

Comment:
What is `"memory"`? Seems only used in error messages and that wouldn't make much sense to the user.

Reply:
I mean the `p_path` parameter in this function is unused, so it should be removed.